### ISSUE TITLE
Using the ONEORMORE : "+" rule with a group throws `ArgumentException`

### ISSUE THREAD

**Issue by dalealleshouse** [b3b00/csly#277]
Although the ZEROORMORE(*) rule works as expected, using the ONEORMORE(+) rule with a group generates a `System.ArgumentException`.

As an example, see the code below. Parsing the string `foo or bar or baz` will generate the error `Unhandled exception. System.ArgumentException: Object of type 'System.Collections.Generic.List 1[System.String]' cannot be converted to type 'System.Collections.Generic.List 1[sly.parser.parser.Group 2[Example.SimpleTokens,System.String]]'.`

``` cs
    public enum SimpleTokens
    {
        [Lexeme(GenericToken.Identifier, IdentifierType.AlphaNumericDash)]
        IDENTIFIER,

        [Lexeme(GenericToken.KeyWord, "or")]
        OR,
    }

    public class SimpleParser
    {
        [Production("widget: IDENTIFIER")]
        public string Widget(Token<SimpleTokens> widget)
        {
            return widget.Value;
        }

        [Production("expression: widget (OR [d] widget)+")]
        public string Expression(string widget, List<Group<SimpleTokens, string>> ors)
        {
            return ors.Aggregate($"{widget}", (acc, a) => $"{acc} | {a.Value(0)}");
        }
    }
```

For convience, I created a simple project that demonstrates the error: https://github.com/dalealleshouse/csly.example

**Comment by b3b00** [comment]
Hello @dalealleshouse I Will look at it tomorrow morning.  Tanks for providing à test case

**Comment by dalealleshouse** [comment]
> Hello @dalealleshouse I Will look at it tomorrow morning. Tanks for providing à test case

Thank you!

**Comment by b3b00** [comment]
issue is now fixed with version 2.8.0.5
